gc/base/MemorySubSpace.cpp

```diff
@@ -936,14 +936,20 @@ MM_MemorySubSpace::systemGarbageCollect(MM_EnvironmentBase* env, uint32_t gcCode
 		reportSystemGCEnd(env);
 		env->releaseExclusiveVMAccessForGC();
 
+		bool isCheckpointGC = (J9MMCONSTANT_EXPLICIT_GC_PREPARE_FOR_CHECKPOINT == gcCode);
 
-		if ((J9MMCONSTANT_EXPLICIT_GC_PREPARE_FOR_CHECKPOINT == gcCode)
+		if (isCheckpointGC
 #if defined(OMR_GC_IDLE_HEAP_MANAGER)
 		|| ((J9MMCONSTANT_EXPLICIT_GC_IDLE_GC == gcCode) && (_extensions->gcOnIdle))
 #endif
 		) {
 			OMRPORT_ACCESS_FROM_ENVIRONMENT(env);
 			uint64_t startTime = omrtime_hires_clock();
+
+			if (isCheckpointGC) {
+				_releaseNurseryPages = true;
```

Comment:
It's not good to introduce a scavenger specific flag in base SubSpace class (various collectors subclass this and it would not really make sense for them).

Not that that I like it, but existing way to deal with this is to pass the memory type via API (releaseFreeMemoryPages). For example, we do it in a few APIs like getActualActiveFreeMemorySize. Unfortunately, it means touching 10 or so files (Heap, Space, Subspace base, SubspaceGenerational).